packages/contract-utils/upgradeable-macros/src/derive.rs

```diff
@@ -0,0 +1,98 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{DeriveInput, Ident};
+
+/// Procedural macro implementation for `#[derive(Upgradeable)]`.
+///
+/// This function generates the implementation of the `Upgradeable` trait for a
+/// given contract type, enabling the contract to be upgraded by replacing its
+/// WASM bytecode.
+///
+/// If the contract struct is annotated with the `#[migratable]` attribute, this
+/// macro will also generate an implementation of the `Migratable` trait, wiring
+/// up the migration and rollback logic based on the `MigratableInternal` trait
+/// provided by the user.
```

Comment:
what happens if `#[migratable]` is there, but user does not provide `MigratableInternal`?